test/UnitTests/Microsoft.Testing.Platform.UnitTests/CommandLine/PlatformCommandLineProviderTests.cs

```diff
@@ -0,0 +1,169 @@
+        Assert.AreEqual(isValid, validateOptionsResult.IsValid);
+
+        if (!isValid)
+        {
+            Assert.AreEqual(string.Format(CultureInfo.InvariantCulture, PlatformResources.PlatformCommandLinePortOptionSingleArgument, PlatformCommandLineProvider.PortOptionKey), validateOptionsResult.ErrorMessage);
+        }
+    }
+
+    [Arguments("32", true)]
+    [Arguments("32.32", false)]
+    [Arguments("invalid", false)]
+    public async Task IsValid_If_ClientPort_Is_Integer(string clientPort, bool isValid)
+    {
+        var provider = new PlatformCommandLineProvider();
+        CommandLineOption option = provider.GetCommandLineOptions().First(x => x.Name == PlatformCommandLineProvider.ClientPortOptionKey);
+
+        ValidationResult validateOptionsResult = await provider.ValidateOptionArgumentsAsync(option, [clientPort]).ConfigureAwait(false);
+        Assert.AreEqual(isValid, validateOptionsResult.IsValid);
+
+        if (!isValid)
+        {
+            Assert.AreEqual(string.Format(CultureInfo.InvariantCulture, PlatformResources.PlatformCommandLinePortOptionSingleArgument, PlatformCommandLineProvider.ClientPortOptionKey), validateOptionsResult.ErrorMessage);
+        }
+    }
+
+    [Arguments("32", true)]
+    [Arguments("32.32", false)]
+    [Arguments("invalid", false)]
+    public async Task IsValid_If_ExitOnProcessExit_Is_Integer(string pid, bool isValid)
+    {
+        var provider = new PlatformCommandLineProvider();
+        CommandLineOption option = provider.GetCommandLineOptions().First(x => x.Name == PlatformCommandLineProvider.ExitOnProcessExitOptionKey);
+
+        ValidationResult validateOptionsResult = await provider.ValidateOptionArgumentsAsync(option, [pid]).ConfigureAwait(false);
+        Assert.AreEqual(isValid, validateOptionsResult.IsValid);
+
+        if (!isValid)
+        {
+            Assert.AreEqual(string.Format(CultureInfo.InvariantCulture, PlatformResources.PlatformCommandLineExitOnProcessExitSingleArgument, PlatformCommandLineProvider.ExitOnProcessExitOptionKey), validateOptionsResult.ErrorMessage);
+        }
+    }
+
+    public async Task IsValid_If_Diagnostics_Provided_With_Other_Diagnostics_Provided()
+    {
+        var provider = new PlatformCommandLineProvider();
+        var options = new Mock<ICommandLineOptions>();
+        _ = options.Setup(o => o.IsOptionSet(PlatformCommandLineProvider.DiagnosticOptionKey)).Returns(true);
+        _ = options.Setup(o => o.IsOptionSet(PlatformCommandLineProvider.DiagnosticOutputDirectoryOptionKey)).Returns(true);
+        _ = options.Setup(o => o.IsOptionSet(PlatformCommandLineProvider.DiagnosticOutputFilePrefixOptionKey)).Returns(true);
+
+        ValidationResult validateOptionsResult = await provider.ValidateCommandLineOptionsAsync(options.Object).ConfigureAwait(false);
+        Assert.IsTrue(validateOptionsResult.IsValid);
+    }
+
+    public async Task IsValid_If_AllDiagnostics_Missing()
```

Comment:
Missing is having a "negative" connotation to me (e.g. something expected is missing). I'd prefer a naming like `WhenNoOptionSpecified`.

Reply:
updated,